cmd/metalctl/app/move.go

```diff
@@ -0,0 +1,250 @@
+
+	for _, field := range []string{"creationTimestamp", "resourceVersion", "uid", "generation", "managedFields"} {
+		delete(so["metadata"].(map[string]any), field)
+	}
+
+	if so["status"] != nil && so["status"].(map[string]any)["conditions"] != nil {
+		for _, field := range so["status"].(map[string]any)["conditions"].([]interface{}) {
+			delete(field.(map[string]any), "lastTransitionTime")
+		}
+	}
+
+	return so
+}
+
+func getCrsToBeMoved(ctx context.Context, targetClient client.Client, sourceCrs []*unstructured.Unstructured) ([]*unstructured.Unstructured, error) {
+	crsToMove := make([]*unstructured.Unstructured, 0, len(sourceCrs))
+	for _, sourceCr := range sourceCrs {
+		targetCr := sourceCr.DeepCopy()
+		err := targetClient.Get(ctx, client.ObjectKeyFromObject(sourceCr), targetCr)
+		if apierrors.IsNotFound(err) {
+			crsToMove = append(crsToMove, sourceCr)
+			continue
+		}
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to check CR existence in the target cluster: %w", err)
+		}
+
+		if reflect.DeepEqual(clearFields(sourceCr), clearFields(targetCr)) {
+			slog.Debug("source and target CRs are the same", slog.String("CR", crName(sourceCr)))
+			continue
+		}
+		return nil, fmt.Errorf("a CR %s/%s already exists in the target cluster and is different then in the source cluster", sourceCr.GetNamespace(), sourceCr.GetName())
+	}
+	return crsToMove, nil
+}
+
+type Node struct {
+	Cr       *unstructured.Unstructured
+	Children []*Node
+}
+
+func crsOwnerReferenceTrees(crs []*unstructured.Unstructured) []*Node {
+	nodeMap := make(map[types.UID]*Node)
+
+	for _, cr := range crs {
+		nodeMap[cr.GetUID()] = &Node{Cr: cr}
+	}
+	roots := []*Node{}
+	for _, cr := range crs {
+		if len(cr.GetOwnerReferences()) == 0 || nodeMap[cr.GetOwnerReferences()[0].UID] == nil {
+			roots = append(roots, nodeMap[cr.GetUID()])
+		} else {
+			owner := nodeMap[cr.GetOwnerReferences()[0].UID]
+			owner.Children = append(owner.Children, nodeMap[cr.GetUID()])
+		}
+	}
+	return roots
+}
+
+func cleanup[T client.Object](ctx context.Context, cl client.Client, objs []T) error {
+	cleanupErrs := make([]error, 0)
+	for _, obj := range objs {
+		if err := cl.Delete(ctx, obj); err != nil {
+			cleanupErrs = append(cleanupErrs, err)
+		}
+	}
+	return errors.Join(cleanupErrs...)
+}
+
+func moveCrs(ctx context.Context, cl client.Client, crsTrees []*Node, ownerUid ...types.UID) (movedCrs []*unstructured.Unstructured, err error) {
+	movedCrs = make([]*unstructured.Unstructured, 0)
+
+	for _, crsTree := range crsTrees {
+		ownerReferences := crsTree.Cr.GetOwnerReferences()
+		if len(ownerReferences) == 1 && len(ownerUid) == 1 {
+			ownerReferences[0].UID = ownerUid[0]
+			crsTree.Cr.SetOwnerReferences(ownerReferences)
+		}
+
+		crsTree.Cr.SetResourceVersion("")
+		if err = cl.Create(ctx, crsTree.Cr); err != nil {
+			err = fmt.Errorf("CR %s couldn't be created in the target cluster: %w", crName(crsTree.Cr), err)
+			return
+		}
```

Comment:
I think the status subresource creation is missing.

Reply:
I have added status update after getting CR however in tests `cl.Create` creates object with status so I'm not sure if it will work on real environment. It would be good to test using clusters.